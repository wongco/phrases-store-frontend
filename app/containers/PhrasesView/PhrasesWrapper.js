import styled from 'styled-components';

const PhraseWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  width: 80%;
`;

export default PhraseWrapper;
