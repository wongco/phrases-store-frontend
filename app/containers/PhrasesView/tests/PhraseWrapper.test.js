import React from 'react';
import { shallow, mount } from 'enzyme';
import { enzymeFind } from 'styled-components/test-utils';
import toJson from 'enzyme-to-json';
import PhrasesWrapper from '../PhrasesWrapper';

describe('<PhrasesWrapper />', () => {
  it('renders without crashing', () => {
    shallow(<PhrasesWrapper />);
  });

  it('matches snapshot for default criteria', () => {
    const wrapper = shallow(<PhrasesWrapper />);
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });

  it('should render a prop', () => {
    const id = 'testId';
    const wrapper = mount(<PhrasesWrapper id={id} />);
    const renderedComponent = enzymeFind(wrapper, PhrasesWrapper);
    expect(renderedComponent.prop('id')).toBe('testId');
  });

  it('should render its text', () => {
    const children = 'Text';
    const wrapper = mount(<PhrasesWrapper>{children}</PhrasesWrapper>);
    const renderedComponent = enzymeFind(wrapper, PhrasesWrapper);
    expect(renderedComponent.contains(children)).toBe(true);
  });
});
