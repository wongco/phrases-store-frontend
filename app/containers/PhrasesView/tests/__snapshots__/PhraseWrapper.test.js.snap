// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhrasesWrapper /> matches snapshot for default criteria 1`] = `
<StyledComponent
  forwardedClass={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "PhrasesWrapper__PhraseWrapper-sc-5kuq2w-0",
        "isStatic": true,
        "rules": Array [
          "display:flex;flex-direction:column;align-items:flex-start;width:80%;",
        ],
      },
      "displayName": "PhrasesWrapper__PhraseWrapper",
      "render": [Function],
      "styledComponentId": "PhrasesWrapper__PhraseWrapper-sc-5kuq2w-0",
      "target": "div",
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
